File: backend/models/studentReport.go
package models

import "time"

type StudentReport struct {
	StudentReportId int         `json:"student_report_id" db:"student_report_id"`
	ReportTypeId    int         `json:"-" db:"report_type_id"`
	ReportType      *ReportType `json:"report_type,omitempy"`
	StudentId       int         `json:"student_id" db:"student_id"`
	DateReported    time.Time   `json:"date_reported" db:"date_reported"`
}

type StudentReportRepository interface {
	GetByStudentIdInDay(int, time.Time) (*[]StudentReport, error)
	Add(StudentReport) (*StudentReport, error)
	DeleteByIdAndDate(int, time.Time) (*StudentReport, error)
}

File: App/lib/data/models/student_report.dart
import 'dart:convert';

import 'package:equatable/equatable.dart';

import 'package:jedi/data/models/report_type.dart';
import 'package:intl/intl.dart';

class StudentReport extends Equatable {
  final int? id;
  final int studentId;
  final ReportType reportType;
  final DateTime dateReported;

  StudentReport({
    this.id,
    required this.studentId,
    required this.reportType,
    required this.dateReported,
  });

  @override
  List<Object?> get props => [id, studentId, reportType, dateReported];

  Map<String, dynamic> toJson() {
    return {
      'student_report_id': id,
      'student_id': studentId,
      'report_type': reportType.toJson(),
      'date_reported': DateFormat('y-M-d').format(dateReported),
    };
  }

  factory StudentReport.fromJson(Map<String, dynamic> map) {
    return StudentReport(
      id: map['student_report_id'],
      studentId: map['student_id'],
      reportType: ReportType.fromJson(map['report_type']),
      dateReported: DateFormat('y-M-d').parse(map['date_reported']),
    );
  }
}
